staging-service/src/main/java/com/github/cafdataprocessing/services/staging/DiskAccessHealthIndicatorWithTimeout.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019-2022 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+
+/**
+ *
+ * @author TBroadbent
+ */
+final class DiskAccessHealthIndicatorWithTimeout extends AbstractHealthIndicator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(DiskAccessHealthIndicatorWithTimeout.class);
+
+    private final File healthcheckFile;
+    private final int healthcheckTimeoutSeconds;
+    private final ExecutorService healthcheckExecutor;
+
+    public DiskAccessHealthIndicatorWithTimeout(
+        final File path, final int healthcheckTimeoutSeconds)
+    {
+        super();
+        this.healthcheckFile = new File(path.getPath() + File.separator + "healthcheck-file.txt");
```

Comment:
Don't use the legacy file I/O packages in new code.  Try to transition to the new `java.nio.file` package and then use the functionality from it.  Revert to the legacy `java.io.File` as little as possible.

So make `healthcheckFile` a `Path` rather than a `File`, and change this line to `this.healthcheckFile = path.toPath().resolve("healthcheck-file.txt");`

See [https://docs.oracle.com/javase/tutorial/essential/io/legacy.html](https://docs.oracle.com/javase/tutorial/essential/io/legacy.html).